AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -631,8 +647,65 @@ public protocol AWSAppSyncOfflineMutationDelegate {
     func mutationCallback(recordIdentifier: String, operationString: String, snapshot: Snapshot?, error: Error?) -> Void
 }
 
+public struct AppSyncConnectionInfo {
+    public let isConnectionAvailable: Bool
+    public let isInitialConnection: Bool
+}
+
+internal extension Notification.Name {
+    internal static let appSyncReachabilityChanged = Notification.Name("AppSyncNetworkAvailabilityChangedNotification")
+}
+
+class AWSAppSyncNetworkStatusChangeNotifier {
+    var reachability: Reachability?
+    var allowsCellularAccess: Bool = true
+    var isInitialConnection: Bool = true
+    
+    static func setupSharedInstance(host: String, allowsCellular: Bool) {
+        sharedInstance = AWSAppSyncNetworkStatusChangeNotifier(host: host, allowsCellular: allowsCellular)
+    }
+    
+    static var sharedInstance: AWSAppSyncNetworkStatusChangeNotifier?
+    
+    init(host: String, allowsCellular: Bool) {
+        reachability = Reachability(hostname: host)
+        allowsCellularAccess = allowsCellular
+        NotificationCenter.default.addObserver(self, selector: #selector(checkForReachability(note:)), name: .reachabilityChanged, object: reachability)
+        do{
+            try reachability?.startNotifier()
+        } catch {
+            
+        }
+    }
+    
+    @objc func checkForReachability(note: Notification) {
+        let reachability = note.object as! Reachability
+        var isReachable = false
+        
+        switch reachability.connection {
+        case .wifi:
+            isReachable = true
+        case .cellular:
+            if (self.allowsCellularAccess) {
+                isReachable = true
+            }
+        case .none:
+            isReachable = false
+        }
+        
+        let info = AppSyncConnectionInfo.init(isConnectionAvailable: isReachable, isInitialConnection: isInitialConnection)
+        
+        if (isInitialConnection) {
```

Comment:
It looks like this should be a `guard !isInitialConnection ...` at the top of the method?

Reply:
Fixed.